test/integration_test.go

```diff
@@ -1333,6 +1333,42 @@ func (ts *IntegrationTestSuite) TestUpdateInfo() {
 	ts.NoError(run.Get(ctx, nil))
 }
 
+func (ts *IntegrationTestSuite) TestUpdateValidatorRejectedFirstWFT() {
+	ctx := context.Background()
+	wfOptions := ts.startWorkflowOptions("test-update-validator-rejected-first-wft")
+	// Add start delay to make sure the update is in the first WFT
+	wfOptions.StartDelay = time.Hour
+	run, err := ts.client.ExecuteWorkflow(ctx,
+		wfOptions, ts.workflows.UpdateWithValidatorWorkflow)
+	ts.Nil(err)
+	// Send a bad update request that will get rejected
+	handler, err := ts.client.UpdateWorkflow(ctx, run.GetID(), run.GetRunID(), "update", "")
+	ts.NoError(err)
+	err = handler.Get(ctx, nil)
+	ts.Error(err)
+	// complete workflow
+	ts.NoError(ts.client.SignalWorkflow(ctx, run.GetID(), run.GetRunID(), "finish", "finished"))
+	ts.NoError(run.Get(ctx, nil))
+}
+
+func (ts *IntegrationTestSuite) TestUpdateValidatorRejected() {
+	ctx := context.Background()
+	wfOptions := ts.startWorkflowOptions("test-update-validator-rejected")
+	run, err := ts.client.ExecuteWorkflow(ctx,
+		wfOptions, ts.workflows.UpdateWithValidatorWorkflow)
+	ts.Nil(err)
+	_, err = ts.client.QueryWorkflow(ctx, run.GetID(), run.GetRunID(), "__stack_trace")
+	ts.NoError(err)
+	// Send a bad update request that will get rejected
+	handler, err := ts.client.UpdateWorkflow(ctx, run.GetID(), run.GetRunID(), "update", "")
+	ts.NoError(err)
+	err = handler.Get(ctx, nil)
+	ts.Error(err)
+	// complete workflow
+	ts.NoError(ts.client.SignalWorkflow(ctx, run.GetID(), run.GetRunID(), "finish", "finished"))
+	ts.NoError(run.Get(ctx, nil))
+}
+
```

Comment:
I'm not quite understanding these tests yet. As I understand it, the bug being fixed affects replay. But these tests don't seem to do anything that would trigger a replay. (I was expecting them to kill and then restart a worker or something, to force it to replay from the start.)

Reply:
All  integration tests are run with and without cache. Running this test without cache will test that it replays deterministically. Explicit replay tests are only needed in rare situations

Reply:
Thanks!